test/compaction_test.go

```diff
@@ -0,0 +1,135 @@
+package test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	. "github.com/onsi/gomega"
+	clientv3 "go.etcd.io/etcd/client/v3"
+)
+
+func TestCompaction(t *testing.T) {
+	ctx := context.Background()
+	client, backend := newKine(t)
+
+	t.Run("SmallDatabaseDeleteEntry", func(t *testing.T) {
+		g := NewWithT(t)
+
+		// Add a few entries
+		numEntries := 2
+		addEntries(ctx, g, client, numEntries)
+
+		// Delete an entry
+		keyNo := 0
+		key := fmt.Sprintf("testkey-%d", keyNo)
+		deleteEntry(ctx, g, client, key)
+
+		initialSize, err := backend.DbSize(ctx)
+		g.Expect(err).To(BeNil())
+
+		err = backend.DoCompact()
+		g.Expect(err).To(BeNil())
+
+		finalSize, err := backend.DbSize(ctx)
+		g.Expect(err).To(BeNil())
+
+		// Expecting no compaction
+		g.Expect(finalSize == initialSize).To(BeTrue())
+
+		// Delete the remaining entry before the next test
+		deleteEntries(ctx, g, client, 1, 2)
+	})
+
+	t.Run("LargeDatabaseDeleteFivePercent", func(t *testing.T) {
+		g := NewWithT(t)
+
+		// Add a large number of entries
+		numAddEntries := 100_000
+		addEntries(ctx, g, client, numAddEntries)
+
+		// Delete 5% of the entries
+		numDelEntries := 5000
+		start := 0
+		deleteEntries(ctx, g, client, start, start+numDelEntries)
+
+		initialSize, err := backend.DbSize(ctx)
+		g.Expect(err).To(BeNil())
+
+		err = backend.DoCompact()
+		g.Expect(err).To(BeNil())
+
+		finalSize, err := backend.DbSize(ctx)
+		g.Expect(err).To(BeNil())
+
+		// Expecting compaction
+		g.Expect(finalSize < initialSize).To(BeTrue())
```

Comment:
Improves the output in case the test case fails 
```suggestion
		g.Expect(finalSize).To(BeNumerically("<", initialSize)
```